### ISSUE TITLE
onChange event not firing, can't type in ChipInput

### ISSUE THREAD

**Issue by gknapp** [TeamWertarbyte/material-ui-chip-input#164]
I've wrapped `ChipInput` and to make it a controlled input. I've also tried uncontrolled but I don't see any redux actions being created after the focus action. Typing in the field does nothing and I see no actions in the console / redux-logger.

I've tried overriding `onChange` (when uncontrolled and using `defaultValue`), switching this to a stateful component as per the controlled input example - same issue.

```javascript
import PropTypes from 'prop-types';
import React from 'react';
import ChipInput from 'material-ui-chip-input';
import { injectIntl } from 'react-intl';

const MaterialChipInput = ({ input, label, meta: { touched, error }, ...custom }) => {
  /*  Remove intl from custom props, if passed down to <input /> we'll get a warning */
  const { intl, ...restCustom } = custom;

  return (
    <ChipInput
      {...input}
      value={input.value || []}
      onRequestAdd={(addedChip) => {
        const values = input.value || [];
        input.onChange([...values, addedChip]);
      }}
      onRequestDelete={(deletedChip) => {
        const values = input.value || [];
        input.onChange(values.filter(v => v !== deletedChip));
      }}
      floatingLabelText={label}
      errorText={touched && error &&
        intl.formatMessage({
          id: error.translateString.id,
          defaultMessage: error.translateString.defaultMessage
        }, {
          validateValue: error.validateValue || ''
        })}
      {...restCustom}
    />
  );
};

MaterialChipInput.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.object
  }).isRequired,
  intl: PropTypes.object.isRequired
};

export default injectIntl(MaterialChipInput);
```

Included in form:

```jsx
<Field
      name="emailAddresses"
      component={MaterialChipInput}
      label={intl.formatMessage({
        id: 'notification.dialog.input.custom',
        defaultMessage: 'Custom Email Addresses'
      })}
      hintText={intl.formatMessage({
        id: 'notification.dialog.input.custom.hint',
        defaultMessage: 'Type the email address and press ENTER'
      })}
      fullWidth
      type="email"
      disabled={!isEnabled || defaultEmails}
    />
```

I'm struggling to debug it as nothing happens on key input after entering the field.

Any ideas? `TextField`, `Checkbox` and other components on the form work fine, just `ChipInput` that you can't type into.

TIA

**Comment by leMaik** [comment]
`onRequestAdd` should be called after pressing enter :thinking: Could you add some log statements to investigate this?

**Comment by espinacs** [comment]
Hi @leMaik and @gknapp , i've been having the same problem (nothing happens on key input) with the new release (v0.18.3), event with the minimum configuration
`<ChipInput
    hintText='Hint text'
  />`

Downgraded to v0.18.2 and everything works fine again.

**Comment by leMaik** [comment]
@espinacs Your example works fine on my machine, with v0.18.3

**Comment by gknapp** [comment]
I was on version `0.15.3`. I upgraded to the latest `0.18.3` release, the problem remained. I just downgraded, as @espinacs did, to `0.18.2` and now I can type in the `ChipInput` again and its working fine.

Could the [`handleAddChip`](https://github.com/TeamWertarbyte/material-ui-chip-input/compare/v0.18.2...v0.18.3) change in `0.18.3` be the cause?

`Field` is from `redux-form`:

```javascript
import { Form, Field, reduxForm } from 'redux-form';
```

Form code:

```javascript
import PropTypes from 'prop-types';
import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';

import MaterialTextField from './../../components/form-fields/MaterialTextField';
import MaterialChipInput from './../../components/form-fields/MaterialChipInput';

const NotificationForm = ({ handleSubmit, onSubmit, intl }) => (
  <Form onSubmit={handleSubmit(onSubmit)}>
    <Field
      name="emailAddresses"
      component={MaterialChipInput}
      label={intl.formatMessage({
        id: 'notification.dialog.input.use-default',
        defaultMessage: 'Use Default Email Addresses'
      })}
      hintText={intl.formatMessage({
        id: 'notification.dialog.input.custom.hint',
        defaultMessage: 'Type the email address and press ENTER'
      })}
      fullWidth
      type="email"
    />
    { /* other fields */ }
  </Form>
);

NotificationForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired
};

const mapStateToProps = (state, ownProps) => {
  const hasData = ownProps.properties && Object.keys(ownProps.properties).length > 0 && ownProps.properties.data;

  return {
    initialValues: hasData ? {
      emailAddresses: ownProps.properties.data.emailAddresses
    } : {}
  };
};

const form = reduxForm({
  form: 'NotificationForm'
});

export default injectIntl(connect(
  mapStateToProps
)(form(NotificationForm)));
```

**Comment by leMaik** [comment]
@gknapp Looks like you digged a bit in the code. :+1: Would you like to try to fix this and submit a PR? That'd be awesome!

**Comment by gknapp** [comment]
I'll try upgrading to 0.18.3 tomorrow and see if I get the same issue. I don't see how any change in the diff I linked relates to what I encountered. The latest patch is for auto-complete and `onBeforeRequestAdd` returning `false`. I can't reproduce with the stories test harness. Maybe it's a cocktail of things in @espinacs and my application environment.

If I can reproduce it outside of my app, I'll submit a PR.

On a separate note, npm 5 shows some warnings on install about deprecated scripts:

```
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
```
